### DISCUSSION TITLE
SEMI 프로젝트

### DISCUSSION THREAD

**Discussion by joonangler** [3sam5oh/openSearch-mini-project#3] [프로젝트]
## 1. 주제 선정
- 훈련생별로 팀 구성을 하여 3가지 Classification 중 한가지 트랙을 선정
- 해당 Classification 내의 프로젝트 주제를 선정하여 구현

#### 애플리케이션 설계
- 도메인 분석 및 도출
- 클라우드 아키텍처 설계
- 화면 설계 및 API 설계

#### 협업 계획 수립
- Issue, PullRequest Template 작성
- 칸반보드에 Task 등록 및 담당자 설정
- Reviewer 지정 및 Merge Rule 설정
- 브랜치 전략 수립

#### 기능 개발
- 백엔드 핵심 모듈 개발

#### 클라우드 서버 구축
- 프로젝트 주제에 맞는 인프라 구성을 AWS에 구축

### 시스템 및 인프라 구성도
- Deployment Architecture
- Cloud Architecture
- API 명세서
 
## 4. 프로젝트 도메인 구현 계획
- #4  내용 기반 5번 항목 일부 계획 구현 예정
- 카테고리 메뉴 클릭 시 검색 엔진을 통해 찾아진 추천 메뉴를 리스트로 반환

**Comment by jlee38266** [comment]
주제 관련 내용 일부

![image](https://github.com/3sam5oh/searchEngineProject/assets/152199695/17653d79-9bdf-4ede-80a7-fe600cffeaec)

![image](https://github.com/3sam5oh/searchEngineProject/assets/152199695/51d97d9b-d4bd-4bc6-8547-0ad631fec492)
